File: contact-list/src/main/java/com/conatct_list/contact_list/service/ContactService.java
package com.conatct_list.contact_list.service;

import com.conatct_list.contact_list.domain.Contact;
import com.conatct_list.contact_list.domain.User;
import com.conatct_list.contact_list.dto.ContactDto;
import org.springframework.data.domain.Page;
import org.springframework.stereotype.Service;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface ContactService {
    public Page<Contact> getAllContacts(int pageNumber, int pageSize);
    public Contact getContact(Long id);
    public Contact createContact(ContactDto contactDto);

    public void deleteContact(Long id);

    public Contact updateContact(Long id, ContactDto contactDto);
    public String uploadPhoto(Long id, MultipartFile file);
    public String photoFunction(Long id, MultipartFile image);

    public String fileExtension(String filename);
    public List<Contact> searchByName(String name);

}

File: contact-list-frontend/src/services/contactService.js
import { UserAxios } from "./helper";

export const saveContact =async (contact) =>{
    return await UserAxios.post(`/contacts`, contact);
}

export const getContacts =async (page=0, size=12) =>{
    return await UserAxios.get(`/contacts?pageNumber=${page}&pageSize=${size}`);
}

export const getContact =async (id) =>{
    return await UserAxios.get(`/contacts/${id}`);
}

export const updateContact =async (id, contact) =>{
    console.log(contact);
    return await UserAxios.put(`/contacts/${id}`, contact);
}

export const deleteContact =async (id) =>{
    return await UserAxios.delete(`/contacts/${id}`);
}

export const uploadPhoto = async (formData) => {
    return await UserAxios.put(`/contacts/photo`, formData);
}

export const getPhototUrl = async (filename) =>{
    return await UserAxios.get(`/contacts/image/${filename}`)
}

export const getSearchedContacts = async (name) =>{
    return await UserAxios.get(`/contacts/search?name=${name}`);
}